src/p2p/net_node.inl

```diff
@@ -793,9 +800,87 @@ bool node_server<t_payload_net_handler>::run() {
     }
 
     log::info(logcat, "net_service loop stopped.");
+    log_detailed_peer_stats();
     return true;
 }
 //-----------------------------------------------------------------------------------
+template<class t_payload_net_handler>
+void node_server<t_payload_net_handler>::log_detailed_peer_stats() {
+    for (const auto& [peer_id, stats] : peer_stats_map) {
+        log::info(logcat, "Peer ID: {}\n\tConnections: {} success, {} failed\n\tBytes Sent: {}\n\tBytes Received: {}\n\tLast Connected: {}\n\tTotal Connection Time: {} seconds",
+            peer_id, stats.successful_connections, stats.failed_connections, stats.bytes_sent, stats.bytes_received, stats.last_connected_timestamp, stats.total_connection_time);
+    }
+}
+//-----------------------------------------------------------------------------------
+template<class t_payload_net_handler>
+void node_server<t_payload_net_handler>::update_peer_stats(const peerid_type peer_id, bool success, size_t bytes_sent, size_t bytes_received, uint64_t connection_time) {
+    auto& stats = peer_stats_map[peer_id];
+    if (success) {
+        stats.successful_connections++;
+    } else {
+        stats.failed_connections++;
+    }
+    stats.bytes_sent += bytes_sent;
+    stats.bytes_received += bytes_received;
+    stats.total_connection_time += connection_time;
+    stats.last_connected_timestamp = time(nullptr);
+}
+//-----------------------------------------------------------------------------------
+template<class t_payload_net_handler>
+uint64_t node_server<t_payload_net_handler>::calculate_peer_score(const peerid_type peer_id) {
+    constexpr uint64_t SCALE_FACTOR = 1000;
+    constexpr uint64_t NEUTRAL_SCORE = 100;
+
+    auto it = peer_stats_map.find(peer_id);
+    if (it == peer_stats_map.end()) {
+        // Assign a neutral score if no stats are available
+        return NEUTRAL_SCORE;
+    }
+
+    const peer_stats& stats = it->second;
+
+
+    // Base scoring factors
+    uint64_t connection_duration_factor = (stats.total_connection_time / 60) * SCALE_FACTOR; // Normalize by minute
```

Comment:
I think you should cap this so that it has a maximum value that enters the score.  Maybe something like: `min(100.0, stats.total_connection_time / 60.0)` so that a *very* long lived connection that has been there for thousands of hours doesn't get a score value that dwarfs everything else.